solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -1701,7 +1701,7 @@ public void publish(final CoreDescriptor cd, final Replica.State state, boolean
         cd.getCloudDescriptor().setLastPublished(state);
       }
       DocCollection coll = zkStateReader.getCollection(collection);
-      if (forcePublish || sendToOverseer(coll, coreNodeName)) {
```

Comment:
@noblepaul @chatman this change is causing test failures. Following tests are failing. Please do run solr tests as I feel there are some more failures

```
Tests with failures [seed: AB00425489678D50]:
   [junit4]   - org.apache.solr.core.TestCoreContainer.testNoCores
   [junit4]   - org.apache.solr.core.TestCoreContainer.testDeleteBadCores
   [junit4]   - org.apache.solr.cloud.DeleteInactiveReplicaTest.deleteInactiveReplicaTest
   [junit4]   - org.apache.solr.cloud.LegacyCloudClusterPropTest.testCreateCollectionSwitchLegacyCloud
   [junit4]   - org.apache.solr.pkg.TestPackages.testCoreReloadingPlugin
   [junit4]   - org.apache.solr.core.FSPRSTest.testShardSplit
```

Reply:
Sure @hiteshk25